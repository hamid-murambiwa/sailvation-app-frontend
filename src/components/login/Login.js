/* eslint-disable*/
import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom/cjs/react-router-dom.min';
import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
import { fetchAllUsers } from '../../redux/Users/users';
import '../../styles/sign.css'

export default function LoginPage() {
  const dispatch = useDispatch();
  const history = useHistory();
  const users =   useSelector((state) => state.userReducer);
  const [userEmail, setUserEmail] = useState('');
  useEffect(() => {
    if(users.langth !== 0) dispatch(fetchAllUsers());
  }, [dispatch]);
  const routeChange = () => {
    const path = `/cruises/reservation`;  
    history.push(path);
  };
  const popUpWrongEmail = () => {
    const message = document.createElement('p');
    message.innerText = 'Wrong Email';
    const wrapper = document.getElementById('wrong-email');
    return wrapper.appendChild(message);
  }
  const submitHandler = () => {
    users.data.map((user) => {
      if (user.email === userEmail) {
        localStorage.setItem('isLoggedIn', 'true');
        localStorage.setItem('userId', `${user.id}`);
        localStorage.setItem('userType', `${user.admin}`)
        routeChange();
      } else {
        popUpWrongEmail()
      }
    });
  };
  return (
        <div className='container mt-5'>
          <div className="form-container doc-flex"> 
            <form  className='m-4'>
              <div className="form-inner">
                <h2 id='log-title'>Login</h2>
                <div className="form-group mt-4">
                  <label htmlFor="email">
                    Email:
                    <input type="email" name="email" id="email" onChange={(e) => setUserEmail(e.target.value)} value={userEmail} />
                  </label>
                </div>
                <div id="wrong-email"></div>
              <button className='mt-3 btn btn-success' type="submit" onClick={(e) => {
                e.preventDefault();
                submitHandler();
              }}
              >
                Login
              </button>
              </div>
            </form>
            <div className='m-4 mBox'>
              <h2>Don't have a account?</h2>
              <Link to="/signup" className='sign-u'>You can Sign Up Here</Link>
            </div>
          </div>
        </div>
  )
}