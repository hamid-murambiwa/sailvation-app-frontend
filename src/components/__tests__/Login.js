import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom/cjs/react-router-dom.min';
import { Provider } from 'react-redux';
import store from '../__mocks__/configureStore';
import Login from '../login/Login';

describe('Login page is working fine', () => {
  test('renders Login', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
