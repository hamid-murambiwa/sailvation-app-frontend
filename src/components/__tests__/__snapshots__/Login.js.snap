// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page is working fine renders Login 1`] = `
<div
  className="container mt-5"
>
  <div
    className="form-container d-flex"
  >
    <div
      className="m-4"
    >
      <h2>
        Don't you have a account
      </h2>
      <a
        className="mt-3"
        href="/signup"
        onClick={[Function]}
      >
        You can Sign Up Here
      </a>
    </div>
    <form
      className="m-4"
    >
      <div
        className="form-inner"
      >
        <h2>
          Login
        </h2>
        <div
          className="form-group mt-4"
        >
          <label
            htmlFor="email"
          >
            Email:
            <input
              id="email"
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </label>
        </div>
        <div
          id="wrong-email"
        />
        <button
          className="mt-3 btn btn-success"
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
